src/server/controllers/match.controller.js

```diff
@@ -9,76 +11,196 @@ import mongoose from 'mongoose';
  * Format the email body
  */
 
-const matchSchema = mongoose.model('match');
+const Match = mongoose.model('match');
+const User = mongoose.model('user');
+
 const EMAIL_SUBJECT = 'New mentee needs your help!';
 const EMAIL_HTML = '<h1>Hi,</br> new mentee needs your mentoring.</h1>';
 const YODA_ACCOUNT = '"Yoda Service Team" <yoda.mentor.lab@gmail.com>';
 
-let matchData = new matchSchema();
-let sampleFailResult = {
-  successCode: 0,
-  errPoint: null,
-  err: null,
-};
-
 // Send mentoring request pushing Email to mentor(receiver)
-function sendRequestEmail(res, mentor, mentee, content) {
+function sendRequestEmail(res, mentor, mentee, content, callback) {
   let transport
     = mailer.createTransport('smtps://yoda.mentor.lab%40gmail.com:svbootcamp@!@smtp.gmail.com');
   let mailOptions = {
     from: YODA_ACCOUNT,
     to: mentor,
-    replyTo: mentee,
     subject: EMAIL_SUBJECT,
     html: EMAIL_HTML + content,
   };
-
   transport.sendMail(mailOptions, function (err, response) {
     if (err) {
-      sampleFailResult.errPoint = 'RequestMentoring - transport.sendMail';
-      sampleFailResult.err = err;
-      res.json(sampleFailResult);
-    } else {
-      res.json('Success');
+      if (typeof callback === 'function')callback(false);
+    }
+    else {
+      if (typeof callback === 'function')callback(true);
     }
-
     transport.close();
   });
 }
 
 // The mentee sent request to Mentor
 export function requestMentoring(req, res, next) {
-  // Todo:
-  // Get Email address from DB with mentorId and menteeId (ObjectId). It's Email address for test now.
-  // Add data to 'matchData'
+  if (typeof req.session.access_token !== 'undefined'
+    && req.session.access_token === req.body.access_token) {
+    let matchData = req.body;
+    matchData.mentee_id = req.session._id;
+    let match = new Match(matchData);
+    console.log(matchData);
+    User.find({_id: matchData.mentor_id}, (err, mentorDoc) => {  //find mentor_id is existing.
+      if (err) {
+        matchCallback.fail.errPoint = 'RequestMentoring - Error occurred when finding montor_id';
+        matchCallback.fail.err = err;
+        res.json(matchCallback.fail);
+      }
+      else {
+        if (mentorDoc.length === 0) {  //if mentor_id is not existing
+          matchCallback.fail.errPoint = 'RequestMentoring - Cannot found mentor.';
+          matchCallback.fail.err = err;
+          res.json(matchCallback.fail);
+        }
+        else {
+          Match.find({mentor_id: matchData.mentor_id, mentee_id: matchData.mentee_id}, (err, matchDoc) => { //find match is existing
```

Comment:
Re: Seongmin's comment:

You don't have to make this nested if-else blocks. For each error handling case, you can just set the required res.json(mathCallback.fail) and return away. That way your code isn't nested (but with a bunch of error handling blocks up top). E.g.

if (error) {
   //do your thing
   return;
}
if (matchDoc.length !== 0) {
    //do your thing
    return;
}

Also, feel free to refactor out callback methods to outside of these methods, if that makes the code less crazy. 

See: http://thecodebarbarian.com/2015/03/20/callback-hell-is-a-myth